src/lightning/fabric/strategies/fsdp.py

```diff
@@ -342,6 +343,9 @@ def _init_cpu_offload(cpu_offload: Optional[Union[bool, "CPUOffload"]]) -> "CPUO
 
 
 def _optimizer_has_flat_params(optimizer: Optimizer) -> bool:
-    from torch.distributed.fsdp import FlatParameter
+    if _TORCH_GREATER_EQUAL_1_13:
```

Comment:
CHANGELOG mentions 2.x, but here you are checking 1.13. Is this an issue on 2.0+ or on 1.13+? Should it be using `_TORCH_GREATER_EQUAL_2_0` instead?

Reply:
While this is an issue on 2.x, this function runs in the context of 1.12+. The attribute being checked is only present in 1.13+ though so for version 1.12 we need to retain the `FlatParameter` based version of the check.